### PR TITLE
PR for review

### PR DESCRIPTION
**Pull request by caalberts**
@calvintan @cbas

### CODE REVIEW

File: client/components/editor.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    var doc = Documents.findOne();
+    if (doc){
+      return doc._id
+    } else {
+      return undefined
+    }
+  },
+  config: function(){
+    return function(editor){
+      // console.log(editor)
```
Comment (sebdeckers) [inline client/components/editor.js]: Clean up debug logging

File: client/components/editor.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      editor.on("change", function(cm_editor, info){
+        // console.log(cm_editor.getValue())
+        $("#viewer_iframe").contents().find("html").html(cm_editor.getValue())
```
Comment (sebdeckers) [inline client/components/editor.js]: Is it normal to do this via jQuery or should it use data binding? (Not sure, just curious)

File: client/components/editor.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        Meteor.call("addEditingUser")
```
Comment (sebdeckers) [inline client/components/editor.js]: Double quotes a few times in this file. Set up linting?

Comment (caalberts) [inline client/components/editor.js]: :+1: will get to linting, test and CI on meteor. Still familiarising myself with the environment

File: client/components/editor.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      return
+    }
+    eusers = EditingUsers.findOne()
+    if (!eusers){
+      return
+    }
+
+    users = []
+    var i = 0;
+    for (var user_id in eusers.users) {
```
Comment (sebdeckers) [inline client/components/editor.js]: Be careful with `for/in` looping over arrays.
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in
- http://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-such-a-bad-idea

File: index.html [hunk 5]
```diff
@@ -1,9 +1,44 @@
...
+    </div>
+
+    <div class="row">
+      <div class="col-sm-12">
+        {{> editingUsers}}
+      </div>
+    </div>
+  </div> -->
+
+<!-- </body> -->
```
Comment (caalberts) [inline index.html]: @calvintan I broke up the index html into the various components

File: client/components/view-files.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+Template.viewFiles.helpers({
+  files: function () {
+    console.dir('helper this', this.gistId)
```
Comment (sebdeckers) [inline client/components/view-files.js]: Debug logging?

File: client/components/view-files.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    return Files.find({ gistId: this.gistId })
+  }
+})
+Template.viewFiles.events({
+  'submit form': function(event) {
+    event.preventDefault()
+
+    console.log(event.target.fileContent.value)
```
Comment (sebdeckers) [inline client/components/view-files.js]: More logging leftovers?

File: lib/collections.js [hunk 8]
```diff
@@ -1,2 +1,4 @@
-
+Gists = new Mongo.Collection("gists")
+Files = new Mongo.Collection("files")
 Documents = new Mongo.Collection("documents")
+EditingUsers = new Mongo.Collection("editingUsers")
```
Comment (caalberts) [inline lib/collections.js]: @calvintan separated Gists and Files collection due Github's use of '.' in filenames as object keys

File: index.html [hunk 9]
```diff
@@ -1,9 +1,44 @@
-<head>
+<!-- <head>
   <title>demo-code-hangout</title>
 </head>
 
-<body>
-  <h1>Welcome to Code Hangout Demo!</h1>
+<body> -->
```
Comment (sebdeckers) [inline index.html]: Why is all this stuff commented out?

File: server/server.js [hunk 10]
```diff
@@ -29,3 +27,79 @@ Meteor.publish('userData', function(){
...
+    const url = 'https://api.github.com/gists/' + id
+    const opts = {
+      headers: { Authorization: 'token ' + Meteor.user().services.github.accessToken }
+    }
+    console.log('fetching ', url, opts)
+    fetch(url, opts)
+      .then(res => res.json())
+      .then(gist => {
+        console.log('received gist')
+        const obj = {
```
Comment (sebdeckers) [inline server/server.js]: Generic variable name

File: server/server.js [hunk 11]
```diff
@@ -29,3 +27,79 @@ Meteor.publish('userData', function(){
...
+          ownerId: Meteor.userId(),
+          files: Object.keys(gist.files),
+          created_at: gist.created_at,
+          updated_at: gist.updated_at
+        }
+        // save each gist into db
+        Gists.upsert({ gistId: id }, obj)
+        // save each file into db
+        Object.keys(gist.files).forEach(file => {
+          const fileObj = Object.assign(gist.files[file], { gistId: id })
```
Comment (sebdeckers) [inline server/server.js]: Mutating input

File: server/server.js [hunk 12]
```diff
@@ -29,3 +27,79 @@ Meteor.publish('userData', function(){
...
+    var doc, user, eusers
+    doc = Documents.findOne()
+    if (!doc){
+      return
+    }
+    if (!this.userId){
+      return
+    }
+    user = Meteor.user().profile
+    eusers = EditingUsers.findOne({docid:doc._id})
```
Comment (sebdeckers) [inline server/server.js]: Inconsistent white spacing

File: server/server.js [hunk 13]
```diff
@@ -29,3 +27,79 @@ Meteor.publish('userData', function(){
...
+    if (!eusers) {
+      eusers = {
```
Comment (sebdeckers) [inline server/server.js]: Avoid variable re-assignment. Can use `||` instead?

File: client/components/actions.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+  results: function () {
+    return Session.get('searchUsers')
+  }
+})
+
+Template.actions.events({
+  // search github users
+  'keyup .searchUser': _.debounce(function (event) {
+    searchGithubUsers(event.target.value)
+  }, 500),
```
Comment (caalberts) [inline client/components/actions.js]: @calvintan you can use something like `_.debounce` or `_.throttle` on keyboard events to periodically save the text from editor to the database

**Pull request merged**